packages/dynamite/dynamite/lib/src/models/type_result/some_of.dart

```diff
@@ -49,13 +51,42 @@ abstract class TypeResultSomeOf extends TypeResult {
 
   String get _typeName {
     final buffer = StringBuffer(r'$');
-    for (final type in subTypes) {
+    for (final type in optimizedSubTypes) {
       buffer.write(toDartName(type.className, uppercaseFirstCharacter: true));
     }
 
     return buffer.toString();
   }
 
+  BuiltList<TypeResult> get _optimizedSubTypes {
+    final optimized = ListBuilder<TypeResult>();
+    final numberCount = subTypes.where(_isNumber).length;
```

Comment:
This variable could be inlined as it is only used once and doesn't make the code less readable imo.